src/query/storage/promremote/storage.go

```diff
@@ -318,23 +307,101 @@ func (p *promStorage) startAsync(_ context.Context) {
 	}()
 }
 
+func deepCopy(queryOpt storage.WriteQueryOptions) storage.WriteQueryOptions {
```

Comment:
hi @hczhu-db , is the deepcopy the fix? suspect the pointer got reset to other values during tenant attribution?

Reply:
Yes, this is the meat of this PR. `wirteQuery` gets reset after `Write()` returns.